service/controller/v22/resource/s3object/current.go

```diff
@@ -18,44 +18,58 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
-
-	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for S3 objects")
-
 	cc, err := controllercontext.FromContext(ctx)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
-
 	bucketName := key.BucketName(customObject, cc.Status.Cluster.AWSAccount.ID)
-	input := &s3.ListObjectsV2Input{
-		Bucket: aws.String(bucketName),
-	}
-	result, err := cc.AWSClient.S3.ListObjectsV2(input)
-	// the bucket can be already deleted with all the objects in it, it is ok if so.
-	if IsBucketNotFound(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "S3 object's bucket not found, no current objects present")
-		return nil, nil
-	} else if err != nil {
-		return nil, microerror.Mask(err)
-	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", "found the S3 objects")
+	var objects []*s3.Object
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding the S3 bucket %#q", bucketName))
+
+		i := &s3.ListObjectsV2Input{
+			Bucket: aws.String(bucketName),
+		}
 
-	output := map[string]BucketObjectState{}
-	for _, object := range result.Contents {
-		body, err := r.getBucketObjectBody(ctx, bucketName, *object.Key)
-		if err != nil {
-			return output, microerror.Mask(err)
+		o, err := cc.AWSClient.S3.ListObjectsV2(i)
+		if IsBucketNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find the S3 bucket %#q", bucketName))
+			return nil, nil
+		} else if err != nil {
+			return nil, microerror.Mask(err)
 		}
 
-		output[*object.Key] = BucketObjectState{
-			Body:   body,
-			Bucket: bucketName,
-			Key:    *object.Key,
+		objects = o.Contents
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found the S3 bucket %#q", bucketName))
+	}
+
+	currentBucketState := map[string]BucketObjectState{}
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding the contents of %d S3 objects", len(objects)))
+
+		for _, object := range objects {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding the content of the S3 object %#q", *object.Key))
```

Comment:
Here is a lot more logging. It would be cool to have some more insights. As we already discussed we could start making more use of different log levels and suppress certain levels in certain environments if we notice we log too much garbage. Here I mainly want to figure how many buckets we have and which ones they are. Next step would be to gather the content in a badge or in parallel, depending on the primitives we have at hand.